usersession/agent/session_agent_test.go

```diff
@@ -103,3 +109,98 @@ func (s *sessionAgentSuite) TestDying(c *C) {
 		c.Error("agent.Dying() channel was not closed when agent stopped")
 	}
 }
+
+func (s *sessionAgentSuite) TestExitOnIdle(c *C) {
+	agent, err := agent.New()
+	c.Assert(err, IsNil)
+	agent.IdleTimeout = 100 * time.Millisecond
+	startTime := time.Now()
+	agent.Start()
+	defer agent.Stop()
+
+	makeRequest := func() {
+		response, err := s.client.Get("http://localhost/v1/session-info")
+		c.Assert(err, IsNil)
+		defer response.Body.Close()
+		c.Check(response.StatusCode, Equals, 200)
+	}
+	makeRequest()
+	time.Sleep(25 * time.Millisecond)
+	makeRequest()
+
+	select {
+	case <-agent.Dying():
+	case <-time.After(2 * time.Second):
+		c.Fatal("agent did not exit after idle timeout expired")
+	}
+	elapsed := time.Now().Sub(startTime)
+	if elapsed < 125*time.Millisecond || elapsed > 175*time.Millisecond {
+		// The idle timeout should have been extended when we
+		// issued a second request after 25ms.
+		c.Errorf("Expected ellaped time close to 125 ms, but got %v", elapsed)
+	}
```

Comment:
This is a good test. Wondering if 50ms is long enough to die for if something went wrong on a very taxed system...